cli/pcluster/config/validators.py

```diff
@@ -1253,3 +1253,25 @@ def fsx_ignored_parameters_validator(section_key, section_label, pcluster_config
 def _describe_ec2_key_pair(key_pair_name):
     """Return information about the provided ec2 key pair."""
     return boto3.client("ec2").describe_key_pairs(KeyNames=[key_pair_name])
+
+
+def ebs_volume_iops_validator(section_key, section_label, pcluster_config):
+    errors = []
+    warnings = []
+
+    section = pcluster_config.get_section("ebs", section_label)
+    volume_size = float(section.get_param_value("volume_size"))
+    volume_type = section.get_param_value("volume_type")
+    volume_type_to_iops_ratio = {"io1": 50, "io2": 500}
+    if section.get_param_value("volume_iops"):
+        volume_iops = float(section.get_param_value("volume_iops"))
+
+    if volume_type == "io1" or volume_type == "io2":
+        if volume_iops < 100 or volume_iops > 64000:
+            errors.append("The EBS volume can be provisioned with from 100 IOPS up to 64,000 IOPS per volume")
+        if volume_iops > volume_size * volume_type_to_iops_ratio[volume_type]:
+            errors.append(
+                "IOPS to volume size ratio of {0} is too high; maximum is 50.".format(volume_iops / volume_size)
```

Comment:
cast to float here since this is the only place you do division. also, use the max value from the dict instead of hard-coding 50 (which i think only applies to io1).

Reply:
Yeah! For io2 it should be 500. Done!